edb/pgsql/compiler/expr.py

```diff
@@ -599,6 +599,17 @@ def _compile_set(
             shape=[expr for expr, _ in ir_set.shape],
             ctx=ctx,
         )
+    elif ir_set.shape and ir_set in ctx.shapes_needed_by_dml:
+        # hoo, boy.
```

Comment:
Please elaborate :-)

Reply:
oops I thought I had gotten rid of this one